### ISSUE TITLE
Middlewares seems to  influence other http applications.

### ISSUE THREAD

**Issue by dylandoamaral** [zio/zio-http#1860]
**Describe the bug**
I don't know if it is a bug or a feature but middlewares seems to influence other http applications.

It is an issue in my case since I have a fallback function to send a custom 404 mesage.

**To Reproduce**
If we take the following code:
```

val app1 = ???
val app2 = ???
val app3 = ???

val app = app1 ++ app2 @@ bearerAuth(token => true) ++ app3
```

The app3 will also have the bearer middleware but not the app1.

**Expected behaviour**
app3 doesn't need token.

**Comment by jdegoes** [comment]
This is a bug. It will be fixed before RC1.

**Comment by jdegoes** [comment]
Closed in favor of #1871, which describes the underlying cause.

**Comment by tusharmath** [comment]
Check my workaround here — https://github.com/zio/zio-http/issues/1627#issuecomment-1370522353
